src/runtime/singlethreaded.rs

```diff
@@ -77,10 +77,40 @@ impl SingleThreaded {
                     let mut executor = executor.enter(enter);
                     // Run the provided future
                     executor.block_on(f).unwrap();
+                });
+            });
+        });
+    }
+
+    /// Blocks and runs all the other futures that are still left in the executor
+    ///
+    /// This is similar to running a runtime, but uses only the current thread.
+    fn shutdown_on_idle(&mut self) -> () {
```

Comment:
In `CurrentThread` this is called just `run`. The `shutdown_on_idle` in the other `Runtime` has the effect that (when running in the background!) it shut downs later on idle.

I would just call this `run` here

Reply:
You also probably want to return the result of `executor.run()`

Reply:
This should IMHO be merged with the `run` function directly (and return errors, etc)

Reply:
I.e. there is no advantage/difference of calling `block_on` in `run` from what I can see?

Reply:
block_on does not call `executor.run().unwrap();`

Reply:
Yes exactly my point. `run` calls `executor.run()`, `block_on` calls `executor.block_on()`. It seems to make no sense to call both one after another in `run`